back_end/src/models/four_year_plan.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+from typing import List
+
+from ..setup import db
+
+class FourYearPlan(db.Model):
+    tableName = "FourYearPlan"
+
+    # TODO: Check nullable and foreign key class names
+    id = db.Column(db.Integer, primary_key=True)
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable = False)
```

Comment:
We need to do `Table_name.id` to establish foreign key. If you want to use lower case, then we should set the table name to be lower case. But if you use Camelcase like in dbdesigner, this should be `User.id`, `AllClasses.id` and `ClassSchedule.id`.